ui/server/src/main/scala/pl/touk/nussknacker/ui/service/ProcessToolbarService.scala

```diff
@@ -0,0 +1,119 @@
+
+object ToolbarPanel {
+
+  import ToolbarHelper._
+
+  def apply(`type`: ToolbarPanelType, title: Option[String], buttonsVariant: Option[ToolbarButtonVariant], buttons: Option[List[ToolbarButton]]): ToolbarPanel =
+    ToolbarPanel(`type`.toString, title, buttonsVariant, buttons)
+
+  def fromConfig(config: ToolbarPanelConfig, process: BaseProcessDetails[_]): ToolbarPanel =
+    ToolbarPanel(
+      config.identity,
+      config.title.map(t => fillByProcessData(t, process)),
+      config.buttonsVariant,
+      config.buttons.map(buttons =>
+        buttons
+          .filterNot(button => {
+            verifyCondition(button.hide, process)
+          })
+          .map(button => ToolbarButton.fromConfig(button, process))
+      )
+    )
+}
+
+@JsonCodec
+case class ToolbarPanel(id: String, title: Option[String], buttonsVariant: Option[ToolbarButtonVariant], buttons: Option[List[ToolbarButton]])
+
+object ToolbarButton {
+
+  import ToolbarHelper._
+
+  def fromConfig(config: ToolbarButtonConfig, process: BaseProcessDetails[_]): ToolbarButton = ToolbarButton(
+    config.`type`,
+    config.title.map(t => fillByProcessData(t, process)),
+    config.icon.map(i => fillByProcessData(i, process)),
+    config.templateHref.map(th => fillByProcessData(th, process)),
+    disabled = verifyCondition(config.disabled, process)
+  )
+}
+
+@JsonCodec
+case class ToolbarButton(`type`: ToolbarButtonType, title: Option[String], icon: Option[String], url: Option[String], disabled: Boolean)
+
+private [service] object ToolbarHelper {
+
+  def createProcessToolbarUUID(baseCode: String, process: BaseProcessDetails[_]): UUID =
+    UUID.nameUUIDFromBytes(s"$baseCode-${process.isSubprocess}-${process.isArchived}".getBytes())
+
+  def fillByProcessData(text: String, process: BaseProcessDetails[_]): String =
+    text
+      .replace("{{processName}}", process.name)
+      .replace("{{processId}}", process.processId.value.toString)
+
+  def verifyCondition(condition: Option[ToolbarCondition], process: BaseProcessDetails[_]): Boolean = {
+    condition.nonEmpty && condition.exists(con => {
+      if (con.shouldVerifyAllProperties) {
+        verifySubprocessCondition(con, process) && verifyArchivedCondition(con, process)
+      } else {
+        verifySubprocessCondition(con, process) || verifyArchivedCondition(con, process)
+      }
+    })
+  }
+
+  private def verifySubprocessCondition(condition: ToolbarCondition, process: BaseProcessDetails[_]) =
+    verifyCondition(process.isSubprocess, condition.subprocess, condition.`type`)
+
+  private def verifyArchivedCondition(condition: ToolbarCondition, process: BaseProcessDetails[_]) =
+    verifyCondition(process.isArchived, condition.archived, condition.`type`)
+
+  private def verifyCondition(toVerify: Boolean, expected: Option[Boolean], `type`: Option[ToolbarConditionType]): Boolean =
+    (`type`.exists(ToolbarConditionType.isAll) && expected.isEmpty) || expected.exists(_.equals(toVerify))
```

Comment:
I don't understand this place. 
{ archived: true, type: allof} - for subprocess it will give true?

Reply:
It depends :) if subprocess is archived ten it will be true otherwise it will be false, because second condition (subprocess) is not configured - so we ignore it.